sqlgen/db_generator.go

```diff
@@ -16,6 +16,11 @@ func (s *State) GenNewTable() *Table {
 	tblName := fmt.Sprintf("tbl_%d", id)
 	newTbl := &Table{ID: id, Name: tblName}
 	newTbl.Collate = CollationType(rand.Intn(int(CollationTypeMax)-1) + 1)
+	if !s.ExistsConfig(ConfigKeyUnitCharsetAndCollationCombine) {
+		newTbl.Charset = newTbl.Collate.GetCharset()
+	} else {
+		newTbl.Charset = CharsetType(rand.Intn(int(CharsetTypeMax)-1) + 1)
+	}
```

Comment:
```suggestion
	newTbl.Charset = newTbl.Collate.GetCharset()
```
It is better to make sure the generated SQLs are semantic-correct. Random charset may cause invalid collation errors like
```
ERROR 1253 (42000): COLLATION 'gbk_bin' is not valid for CHARACTER SET 'utf8mb4'
```